students/thorn/lesson03/mailroom.py

```diff
@@ -26,33 +26,69 @@ def send_thanks():
     - Print thank you note
     - Returns to original prompt or quit to prompt
     """
-    pass
+    name_request = input("Please enter a donor's full name, or 'list' for a full list of donors.  'Quit' or 'Return' valid commands. ")
 
+    # Return or Quit.
+    if name_request.lower() == "quit":
+        quit()
+
+    if name_request.lower() == "return":
+        return
+
+    # List request - only donors returned.
+    if name_request.lower() == 'list':
+        for donors in DONOR_HISTORY:
+            print(donors[0])
+        return
```

Comment:
if / elif here as well

it is more syntactically correct to have `for donor in DONOR_HISTORY` as you are getting a single donor when you're iterating

( i would really call it `for donor in donors`)